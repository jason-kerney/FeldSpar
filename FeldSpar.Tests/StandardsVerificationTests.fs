namespace FeldSpar.Console.Tests
open FeldSpar.Framework
open FeldSpar.Framework.Verification
open ApprovalTests.Reporters

module StandardsVerificationTests =
    type Color = | White | Brown | Black | TooCool
    type TestingType =
        {
            Name : string;
            Age:int;
            Dojo:string*Color
        }
    
    let ``A test to check verification`` = 
        Test((fun env ->
                let itemUnderTest = 
                    sprintf "%A%s"
                        ({
                            Name = "Steven";
                            Age = 38;
                            Dojo = ("Too Cool For School", TooCool)
                        }) "\n"

                verify
                    {
                        let! standardsAreGood = itemUnderTest |> checkAgainstStringStandard env
                        return Success
                    }
            ))

    let ``Can test a query`` = 
        Test(fun env ->
            let getQuery (x: int) = x.ToString ()
            let executeQuery (s: string) = s

            let query = { QueryResult = 1234; GetQuery = getQuery; ExecuteQuery = executeQuery }

            checkQueryResultAgainstStandard env query
        )