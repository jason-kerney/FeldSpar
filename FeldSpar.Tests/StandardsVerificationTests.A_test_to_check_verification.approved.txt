{Name = "Steven";
 Age = 38;
 Dojo = ("Too Cool For School", TooCool);}
